sbin/functions.sh

```diff
@@ -964,6 +964,469 @@ clean_up()
+					return 1
+					;;
+				1)
+					ARTIFACTS_DIR="`pwd`/tmp/deploy/images/`ls tmp/deploy/images/`"
+					;;
+				*)
+					filelist=`ls tmp/deploy/images` menuitems=""
+					for i in $filelist; do
+						menuitems="$menuitems $i $i "
+					done
+					dialog --no-cancel --menu "Found multiple machine target in your project, Please choose one:" \
+						$(expr $filecount + 7) 30 $filecount $menuitems 2>$tmpfile
+					ARTIFACTS_DIR="`pwd`/tmp/deploy/images/$(cat $tmpfile)"
+					;;
+			esac
+		elif [ -e "images" ]; then
+			ARTIFACTS_DIR="`pwd`/images"
+		else
+			# default to current dir
+			ARTIFACTS_DIR="`pwd`"
+		fi
+	fi
+	echo "ARTIFACTS_DIR=\"${ARTIFACTS_DIR}\"" >> ${localconf}
+	filecount=`ls $ARTIFACTS_DIR -1 | grep -v menifest | wc -l`
+	if [ $filecount -lt 3 ]; then
+		debugmsg ${DEBUG_CRIT} "Artifacts are not sufficient. Please verify."
+		return 1
+	fi
+
+	# exclude manifest files
+	filelist=`ls $ARTIFACTS_DIR | grep -v manifest` menuitems=""
+	local checklistmenu=""
+	for i in $filelist; do
+		menuitems="$menuitems $i $i "
+		checklistmenu="$checklistmenu $i $i off "
+	done
+
+	# Kernel selection
+	dialog --no-tags --menu "Please choose kernel:" \
+		$(expr $filecount + 7) 100 $filecount $menuitems 2>$tmpfile
+	dialog_res=$(cat $tmpfile)
+	if [ -z "${dialog_res}" ]; then
+		debugmsg ${DEBUG_CRIT} "Kernel not specified. Please try again."
+		return 1
+	fi
+	echo "INSTALL_KERNEL=\"\${ARTIFACTS_DIR}/${dialog_res}\"" >> ${localconf}
+
+	# Rootfs selection
+	dialog --no-tags --menu "Please choose rootfs:" \
+		$(expr $filecount + 7) 100 $filecount $menuitems 2>$tmpfile
+	dialog_res=$(cat $tmpfile)
+	if [ -z "${dialog_res}" ]; then
+		debugmsg ${DEBUG_CRIT} "Rootfs not specified. Please try again."
+		return 1
+	fi
+	echo "INSTALL_ROOTFS=\"\${ARTIFACTS_DIR}/${dialog_res}\"" >> ${localconf}
+	echo "HDINSTALL_ROOTFS=\"\${ARTIFACTS_DIR}/${dialog_res}\"" >> ${localconf}
+
+	# Initramfs selection
+	dialog --no-tags --menu "Please choose initramfs:" \
+		$(expr $filecount + 7) 100 $filecount $menuitems 2>$tmpfile
+	dialog_res=$(cat $tmpfile)
+	if [ -z "${dialog_res}" ]; then
+		debugmsg ${DEBUG_CRIT} "Initramfs not specified. Please try again."
+		return 1
+	fi
+	echo "INSTALL_INITRAMFS=\"\${ARTIFACTS_DIR}/${dialog_res}\"" >> ${localconf}
+
+	# containers selection
+	if [ -d "$ARTIFACTS_DIR/containers" ]; then
+		filelist=`ls $ARTIFACTS_DIR/containers` checklistmenu="" filecount=0
+		for i in $filelist; do
+			let filecount++
+			checklistmenu="$checklistmenu $i $i off "
+		done
+	fi
+	dialog --no-tags --checklist "Please choose install containers:" \
+		$(expr $filecount + 7) 100 $filecount $checklistmenu 2>$tmpfile
+	dialog_res=$(cat $tmpfile)
+	if [ -z "${dialog_res}" ]; then
+		debugmsg ${DEBUG_CRIT} "Containers not specified. Please try again."
+		return 1
+	fi
+
+	local hdcontainers=${dialog_res} fmtcontainers=""
+
+	# VT and network prime selection
+	menuitems=""
+	filecount=0
+	for i in ${hdcontainers}; do
```

Comment:
The goal of the interactive installer's questions was that they should be scripts that are dectected, sourced and then run. 

To change these, you must change the installer, or it's library routines.

i.e. there should be something like 'prompts/foo.sh", that will be detected, sourced and then executed. These scripts can assume any of the installer variables, and also be passed information in a standard routine like "foo_installer $artefacts ..."